src/main/scala/io/iohk/ethereum/jsonrpc/PersonalService.scala

```diff
@@ -157,7 +157,12 @@ class PersonalService(
       val maybeCurrentNonce = getCurrentAccount(request.from).map(_.nonce.toBigInt)
       val maybeNextTxNonce = maybeLatestPendingTxNonce.map(_ + 1) orElse maybeCurrentNonce
       val tx = request.toTransaction(maybeNextTxNonce.getOrElse(blockchainConfig.accountStartNonce))
-      val stx = wallet.signTx(tx)
+
+      val stx = if(blockchain.getBlockByNumber(blockchainConfig.eip155BlockNumber).isDefined){
+        wallet.signTx(tx, Some(blockchainConfig.chainId))
+      }else{
+        wallet.signTx(tx, None)
+      }
```

Comment:
but what if branch is longer than 1, in that case your previous argument will apply, this is why I proposed to set this to arbitrary configured value instead of `1` which may be as good as 
checking for exact bestBlockNumber / checking if fork block exists